### PR TITLE
Replace Match.allMatchNotEmpty usage

### PR DESCRIPTION
**Pull request by RoiEXLab**
This removes the usage of Match.allMatchNotEmpty.
Most of this PR was done using find-and-replace using regex expressions.
I reviewed every change myself afterwards, but it's very likely I overlooked something...

### CODE REVIEW

File: src/main/java/games/strategy/triplea/delegate/MoveValidator.java [hunk 1]
```diff
@@ -308,7 +308,7 @@ private static MoveValidationResult validateCombat(final GameData data, final Co
     }
     // if there is a neutral in the middle must stop unless all are air or getNeutralsBlitzable
     if (route.hasNeutralBeforeEnd()) {
-      if (!Match.allMatchNotEmpty(units, Matches.UnitIsAir) && !isNeutralsBlitzable(data)) {
+      if (units.isEmpty() || !Match.allMatch(units, Matches.UnitIsAir) && !isNeutralsBlitzable(data)) {
```
Comment (ssoloff) [inline src/main/java/games/strategy/triplea/delegate/MoveValidator.java]: Do we need parentheses here due to `&&` having a higher precedence than `||`?  For example:

```java
(units.isEmpty() || !Match.allMatch(units, Matches.UnitIsAir)) && !isNeutralsBlitzable(data)
```

File: src/main/java/games/strategy/triplea/delegate/MoveValidator.java [hunk 2]
```diff
@@ -447,7 +448,7 @@ private static MoveValidationResult validateNonCombat(final GameData data, final
     }
     // if there are enemy units on the path blocking us, that is validated elsewhere (validateNonEnemyUnitsOnPath)
     // now check if we can move over neutral or enemies territories in noncombat
-    if (Match.allMatchNotEmpty(units, Matches.UnitIsAir)
+    if (!units.isEmpty() && Match.allMatch(units, Matches.UnitIsAir)
```
Comment (ssoloff) [inline src/main/java/games/strategy/triplea/delegate/MoveValidator.java]: Parentheses are not required here, but it may make it easier to read (I got hung up on the `||` for a few minutes).

File: src/main/java/games/strategy/triplea/delegate/MoveValidator.java [hunk 3]
```diff
@@ -642,7 +644,7 @@ private static MoveValidationResult validateBasic(final GameData data,
       }
       // if there is a neutral in the middle must stop unless all are air or getNeutralsBlitzable
       if (route.hasNeutralBeforeEnd()) {
-        if (!Match.allMatchNotEmpty(units, Matches.UnitIsAir) && !isNeutralsBlitzable(data)) {
+        if (units.isEmpty() || !Match.allMatch(units, Matches.UnitIsAir) && !isNeutralsBlitzable(data)) {
```
Comment (ssoloff) [inline src/main/java/games/strategy/triplea/delegate/MoveValidator.java]: Parentheses required?

File: src/main/java/games/strategy/triplea/delegate/MoveValidator.java [hunk 4]
```diff
@@ -1545,8 +1549,9 @@ public static Route getBestRoute(final Territory start, final Territory end, fin
           data.getMap().getRoute_IgnoreEnd(start, end, Match.allOf(Matches.TerritoryIsWater, noImpassable));
       if (waterRoute != null
           && ((waterRoute.getLargestMovementCost(unitsWhichAreNotBeingTransportedOrDependent) <= defaultRoute
-              .getLargestMovementCost(unitsWhichAreNotBeingTransportedOrDependent)) || (forceLandOrSeaRoute && Match
-                  .anyMatch(unitsWhichAreNotBeingTransportedOrDependent, Matches.UnitIsSea)))) {
+              .getLargestMovementCost(unitsWhichAreNotBeingTransportedOrDependent)) || (forceLandOrSeaRoute
+                  && Match
+                      .anyMatch(unitsWhichAreNotBeingTransportedOrDependent, Matches.UnitIsSea)))) {
```
Comment (ssoloff) [inline src/main/java/games/strategy/triplea/delegate/MoveValidator.java]: It might be easier to read if lines 1553 and 1554 were combined (doesn't look like it will exceed max line length).

File: src/main/java/games/strategy/triplea/delegate/MustFightBattle.java [hunk 5]
```diff
@@ -379,15 +379,15 @@ public void fight(final IDelegateBridge bridge) {
       // play a sound
       if (Match.anyMatch(m_attackingUnits, Matches.UnitIsSea)
           || Match.anyMatch(m_defendingUnits, Matches.UnitIsSea)) {
-        if (Match.allMatchNotEmpty(m_attackingUnits, Matches.UnitIsSub)
+        if (!m_attackingUnits.isEmpty() && Match.allMatch(m_attackingUnits, Matches.UnitIsSub)
```
Comment (ssoloff) [inline src/main/java/games/strategy/triplea/delegate/MustFightBattle.java]: Parentheses are not required here, but it may make it easier to read due to the `||`.

File: src/main/java/games/strategy/triplea/delegate/MustFightBattle.java [hunk 6]
```diff
@@ -1201,7 +1203,7 @@ private boolean canAttackerRetreatPartialAmphib() {
     // non-paratrooper non-amphibious
     // land units.
     // If attacker is all planes, just return collection of current territory
-    if (m_headless || Match.allMatchNotEmpty(m_attackingUnits, Matches.UnitIsAir)
+    if (m_headless || !m_attackingUnits.isEmpty() && Match.allMatch(m_attackingUnits, Matches.UnitIsAir)
```
Comment (ssoloff) [inline src/main/java/games/strategy/triplea/delegate/MustFightBattle.java]: Parentheses are not required here, but it may make it easier to read due to the `||`.

File: src/main/java/games/strategy/triplea/ui/MovePanel.java [hunk 7]
```diff
@@ -1264,7 +1264,7 @@ public void mouseMoved(final Territory territory, final MouseDetails me) {
             // is the only one for
             // which the route may actually change much)
             if (unitsThatCanMoveOnRoute.size() < selectedUnits.size() && (unitsThatCanMoveOnRoute.size() == 0
-                || Match.allMatchNotEmpty(unitsThatCanMoveOnRoute, Matches.UnitIsAir))) {
+                || !unitsThatCanMoveOnRoute.isEmpty() && Match.allMatch(unitsThatCanMoveOnRoute, Matches.UnitIsAir))) {
```
Comment (ssoloff) [inline src/main/java/games/strategy/triplea/ui/MovePanel.java]: Parentheses are not required here, but it may make it easier to read due to the `||`.

**Comment by ssoloff** [comment]
@ron-murhammer @DanVanAtta It might be good to get a third set of eyes on this due simply to the fine level of detail required in the review.  All the operators start running together by the end of the review. :smile:

File: src/main/java/games/strategy/triplea/delegate/MustFightBattle.java (hunk 5 above)
Comment (RoiEXLab) [inline src/main/java/games/strategy/triplea/delegate/MustFightBattle.java]: I believe this is the wrong line ^^

**Comment by RoiEXLab** [comment]
@ssoloff Updated!

File: src/main/java/games/strategy/triplea/delegate/MustFightBattle.java (hunk 5 above)
Comment (ssoloff) [inline src/main/java/games/strategy/triplea/delegate/MustFightBattle.java]: Sorry, I should have tagged two lines below so the `||` would have been visible:

```java
if (!m_attackingUnits.isEmpty() && Match.allMatch(m_attackingUnits, Matches.UnitIsSub)
    || (Match.anyMatch(m_attackingUnits, Matches.UnitIsSub)
        && Match.anyMatch(m_defendingUnits, Matches.UnitIsSub))) {
```

File: src/main/java/games/strategy/triplea/ui/MovePanel.java [hunk 8]
```diff
@@ -1265,7 +1265,7 @@ public void mouseMoved(final Territory territory, final MouseDetails me) {
             // is the only one for
             // which the route may actually change much)
             if (unitsThatCanMoveOnRoute.size() < selectedUnits.size() && (unitsThatCanMoveOnRoute.size() == 0
-                || Match.allMatchNotEmpty(unitsThatCanMoveOnRoute, Matches.UnitIsAir))) {
+                || (!unitsThatCanMoveOnRoute.isEmpty() && Match.allMatch(unitsThatCanMoveOnRoute, Matches.UnitIsAir)))) {
```
Comment (ssoloff) [inline src/main/java/games/strategy/triplea/ui/MovePanel.java]: Sorry, I didn't see this after 541a4e3, and it broke the build...  Checkstyle: "Line is longer than 120 characters (found 121)."

**Comment by RoiEXLab** [comment]
@ssoloff Updated!

File: src/main/java/games/strategy/triplea/delegate/MoveValidator.java [hunk 9]
```diff
@@ -133,14 +133,14 @@ public static MoveValidationResult validateMove(final Collection<Unit> units, fi
...
-          + MyFormatter.unitsToTextNoOwner(units));
-      return result;
+        && !getEditMode(data)) {
+      final Collection<Unit> matches = Match.getMatches(units,
+          Matches.unitIsBeingTransportedByOrIsDependentOfSomeUnitInThisList(units, route, player, data, true).invert());
+      if (matches.isEmpty() || !Match.allMatch(matches, Matches.unitIsOwnedBy(player))) {
+        result.setError("Player, " + player.getName() + ", is not owner of all the units: "
+            + MyFormatter.unitsToTextNoOwner(units));
+        return result;
+      }
```
Comment (ron-murhammer) [inline src/main/java/games/strategy/triplea/delegate/MoveValidator.java]: Can someone double/triple check this chunk. Its kind of hard to follow. Might be worth refactoring this a little to clean it up.

**Comment by ron-murhammer** [comment]
I doubled checked everything and this looks good. Merging.
